mathics/core/symbols.py

```diff
@@ -468,30 +382,57 @@ def format(self, evaluation, form, **kwargs) -> "BaseExpression":
         )
         return result
 
-    def is_free(self, form, evaluation) -> bool:
+    def get_atoms(self, include_heads=True):
         """
-        Check if self has a subexpression of the form `form`.
+        Returns a list of atoms that appears in the expression.
         """
-        from mathics.builtin.patterns import item_is_free
+        # Comment @mmatera:
+        # This function is used just in Graphics.apply_makeboxes
+        # to check if a graphics expression is composed just by
+        # real numbers (or integer) or graphics symbols.
+        # Probably, there is a simpler way to implement it without using
+        # this method.
+        return []
 
-        return item_is_free(self, form, evaluation)
+    def get_attributes(self, definitions):
+        return nothing
 
-    def is_inexact(self) -> bool:
-        return self.get_precision() is not None
+    # Probably, this method shouldn't be here.
+    def get_elements(self):
+        return []
 
-    def get_precision(self) -> None:
-        """Returns the default specification for precision in N and other
-        numerical functions.  It is expected to be redefined in those
-        classes that provide inexact arithmetic like PrecisionReal.
+    def has_changed(self, definitions):
+        return True
 
-        Here in the default base implementation, `None` is used to indicate that the
-        precision is either not defined, or it is exact as in the case of Integer. In either case, the
-        values is not "inexact".
+    def get_head(self):
+        return None
 
-        This function is called by method `is_inexact()`.
-        """
+    def get_head_name(self):
+        raise NotImplementedError
+
+    # Compatibily with old code. Deprecated, but remove after a little bit.
+    get_leaves = get_elements
+
+    def get_float_value(self, permit_complex=False):
+        return None
+
+    def get_int_value(self):
         return None
 
+    def get_lookup_name(self):
```

Comment:
This function also doesn't smell quite right for me.

Reply:
I am guessing this should be an `Expression` thing, not something here. @mmatera ?